compiler/rustc_typeck/src/check/intrinsic.rs

```diff
@@ -107,14 +112,25 @@ pub fn intrinsic_operation_unsafety(intrinsic: Symbol) -> hir::Unsafety {
         | sym::variant_count
         | sym::ptr_mask => hir::Unsafety::Normal,
         _ => hir::Unsafety::Unsafe,
+    };
+
+    if has_safe_attr != is_in_list {
+        tcx.sess.err(
+        DiagnosticMessage::Str(format!(
+                "intrinsic safety mismatch between list of intrinsics within the compiler and core library intrinsics for intrinsic `{}`",
+                tcx.item_name(intrinsic_id)
+    )));
```

Comment:
Nit: I have no idea why rustfmt is ok with this 😄 Maybe the macro call as fn arg confuses it? I'd suggest formatting it like this unless rustfmt dislikes it:

```suggestion
            DiagnosticMessage::Str(format!(
                "intrinsic safety mismatch between list of intrinsics within the compiler and core library intrinsics for intrinsic `{}`",
                tcx.item_name(intrinsic_id)
        )));
```